ducktape/tests/runner.py

```diff
@@ -14,212 +14,175 @@
 
 
 import logging
+import multiprocessing
+import os
 import time
-import traceback
+import zmq
 
-from ducktape.tests.result import TestResult, TestResults, IGNORE, PASS, FAIL
-from ducktape.tests.reporter import SingleResultFileReporter
-from ducktape.utils.local_filesystem_utils import mkdir_p
+from ducktape.tests.serde import SerDe
+from ducktape.tests.session import SessionLogger
+from ducktape.command_line.defaults import ConsoleDefaults
+from ducktape.tests.runner_client import run_client
+from ducktape.tests.result import TestResults
 from ducktape.utils.terminal_size import get_terminal_size
+from ducktape.tests.message import Request, ready_reply
+from ducktape.tests.logger import Logger
 
 
-class TestRunner(object):
-    """Abstract class responsible for running one or more tests."""
-    def __init__(self, session_context, tests):
-        self.tests = tests
+class RequestLogger(Logger):
+    def __init__(self, session_context):
         self.session_context = session_context
-        self.results = TestResults(self.session_context)
-
-        self.logger.debug("Instantiating " + self.who_am_i())
 
     @property
-    def cluster(self):
-        return self.session_context.cluster
+    def logger_name(self):
+        # Naming means RequestLogger.logger is a child of SessionLogger.logger
+        # I.e. requests
+        return "%s.requestlogger" % SessionLogger(self.session_context).logger_name
 
-    @property
-    def logger(self):
-        return self.session_context.logger
+    def configure_logger(self):
+        """Log to request_log."""
+        if self.configured:
+            return
 
-    def who_am_i(self):
-        """Human-readable name helpful for logging."""
-        return self.__class__.__name__
+        self._logger.setLevel(logging.DEBUG)
 
-    def run_all_tests(self):
-        raise NotImplementedError()
+        fh_debug = logging.FileHandler(os.path.join(self.session_context.results_dir, "request_log.debug"))
+        fh_debug.setLevel(logging.DEBUG)
+
+        # create formatter and add it to the handlers
+        formatter = logging.Formatter(ConsoleDefaults.SESSION_LOG_FORMATTER)
+        fh_debug.setFormatter(formatter)
+
+        # add the handlers to the logger
+        self._logger.addHandler(fh_debug)
 
 
-class SerialTestRunner(TestRunner):
+class TestRunner(object):
     """Runs tests serially."""
 
     # When set to True, the test runner will finish running/cleaning the current test, but it will not run any more
     stop_testing = False
 
-    def __init__(self, *args, **kwargs):
-        super(SerialTestRunner, self).__init__(*args, **kwargs)
+    def __init__(self, cluster, session_context, session_logger, tests, port=ConsoleDefaults.TEST_DRIVER_PORT):
+        # session_logger, message logger,
+        self.session_logger = session_logger
+        self.request_logger = RequestLogger(session_context).logger
```

Comment:
It just seemed like we were already putting logs from the test runners into the session/test logger anyway. I figured the number of requests here should be minimal and mostly just bookend the test, right?